src/irmin-pack/s.ml

```diff
@@ -74,6 +74,62 @@ module type S = sig
 
   val stats :
     dump_blob_paths_to:string option -> commit:commit -> repo -> unit Lwt.t
+
+  module Snapshot : sig
+    type kinded_hash = Contents of hash * metadata | Node of hash
+    [@@deriving irmin]
+
+    type entry = { name : string; hash : kinded_hash } [@@deriving irmin]
+
+    type inode_tree = {
+      depth : int;
+      length : int;
+      pointers : (int * hash) list;
+    }
+    [@@deriving irmin]
+
+    type v = Inode_tree of inode_tree | Inode_value of entry list
+    [@@deriving irmin]
+
+    type inode = { v : v; root : bool } [@@deriving irmin]
+
+    type t = [ `Inode of inode | `Blob of Backend.Contents.Val.t ]
+    [@@deriving irmin]
+
+    type on_disk = [ `Reuse | `Path of string ]
+    (** The type for specifing an [on_disk] option when traversing a tree. The
+        traversal requires an index to keep track of visited elements.
+
+        - if [on_disk] is not specified, the index is in memory.
+        - if [on_disk] is [`Path path], the index is temporarily store at
+          [path].
+        - if [on_disk] is [`Reuse] the store's index is reused. *)
+
+    val export :
+      ?on_disk:on_disk ->
```

Comment:
Does ``~on_disk:`Reuse`` make sense in the context of export?

Reply:
no, it doesn't, I wanted to reuse the same type for import and export. 
I replaced `on_disk` with `string` for export. I don't like so much, because its less clear, but if you think its better I'll leave it like this.

Reply:
Let's just remove the `` `Reusable `` tag from the current types.

Reply:
I actually meant removing `` `Reusable`` from `export` only, and not from `import`. 

``?on_disk:[ `Path of string ]`` in `export` and ``?on_disk:[ `Path of string | `Reuse ]`` in import.

Reuse was a good feature, I think it should be in this PR. Sorry for the ambiguous message :(